ui/messages/html/parser.go

```diff
@@ -377,20 +399,25 @@ func (parser *htmlParser) Parse(htmlData string) Entity {
 	if bodyNode != nil {
 		return parser.singleNodeToEntity(bodyNode)
 	}
+
 	return parser.singleNodeToEntity(node)
 }
 
 const TabLength = 4
 
 // Parse parses a HTML-formatted Matrix event into a UIMessage.
-func Parse(room *rooms.Room, content *event.MessageEventContent, sender id.UserID, senderDisplayname string) Entity {
+func Parse(prefs *config.UserPreferences, room *rooms.Room, content *event.MessageEventContent, sender id.UserID, senderDisplayname string) Entity {
 	htmlData := content.FormattedBody
+	DataMautrixNoLink := false
+
 	if content.Format != event.FormatHTML {
 		htmlData = strings.Replace(html.EscapeString(content.Body), "\n", "<br/>", -1)
 	}
 	htmlData = strings.Replace(htmlData, "\t", strings.Repeat(" ", TabLength), -1)
-
-	parser := htmlParser{room: room}
+	if strings.Contains(htmlData, "data-mautrix-no-link") {
+		DataMautrixNoLink = true
+	}
```

Comment:
The attribute is per-url, not global. Probably need to add a new helper function:

```go
func (parser *htmlParser) hasAttribute(node *html.Node, attribute string) bool {
	for _, attr := range node.Attr {
		if attr.Key == attribute {
			return true
		}
	}
	return false
}
```

Then just `if !parser.hasAttribute(node, "data-mautrix-no-link") { ... }`